
From: NeilBrown <neilb@cse.unsw.edu.au>

We should set the OPEN_RESULT_LOCKTYPE_POSIX flag in the return to open, to
indicate that we support locking.

Signed-off-by: Andy Adamson <andros@citi.umich.edu>
Signed-off-by: J. Bruce Fields <bfields@citi.umich.edu>
Signed-off-by: Neil Brown <neilb@cse.unsw.edu.au>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/fs/nfsd/nfs4state.c  |    2 +-
 25-akpm/include/linux/nfs4.h |    1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff -puN fs/nfsd/nfs4state.c~nfsd4-set-open_result_locktype_posix-in-open fs/nfsd/nfs4state.c
--- 25/fs/nfsd/nfs4state.c~nfsd4-set-open_result_locktype_posix-in-open	2004-09-23 22:13:23.217709288 -0700
+++ 25-akpm/fs/nfsd/nfs4state.c	2004-09-23 22:13:23.224708224 -0700
@@ -1384,7 +1384,7 @@ out:
 	/*
 	* To finish the open response, we just need to set the rflags.
 	*/
-	open->op_rflags = 0;
+	open->op_rflags = NFS4_OPEN_RESULT_LOCKTYPE_POSIX;
 	if (!open->op_stateowner->so_confirmed)
 		open->op_rflags |= NFS4_OPEN_RESULT_CONFIRM;
 
diff -puN include/linux/nfs4.h~nfsd4-set-open_result_locktype_posix-in-open include/linux/nfs4.h
--- 25/include/linux/nfs4.h~nfsd4-set-open_result_locktype_posix-in-open	2004-09-23 22:13:23.218709136 -0700
+++ 25-akpm/include/linux/nfs4.h	2004-09-23 22:13:23.224708224 -0700
@@ -35,6 +35,7 @@
 #define NFS4_FH_VOL_RENAME		0x0008
 
 #define NFS4_OPEN_RESULT_CONFIRM 0x0002
+#define NFS4_OPEN_RESULT_LOCKTYPE_POSIX 0x0004
 
 #define NFS4_SHARE_ACCESS_READ	0x0001
 #define NFS4_SHARE_ACCESS_WRITE	0x0002
_
